pyrealm/pmodel/new_subdaily.py

```diff
@@ -0,0 +1,285 @@
+r"""The :mod:`~pyrealm.pmodel.subdaily` module provides extensions to the P Model that
+incorporate modelling of the fast and slow responses of photosynthesis to changing
+conditions.
+"""  # noqa: D205, D415
+
+from warnings import warn
+
+import numpy as np
+from numpy.typing import NDArray
+
+from pyrealm import ExperimentalFeatureWarning
+from pyrealm.pmodel import (
+    FastSlowScaler,
+    PModel,
+    PModelEnvironment,
+    calc_ftemp_arrh,
+    calc_ftemp_kphio,
+    memory_effect,
+)
+from pyrealm.pmodel.optimal_chi import OPTIMAL_CHI_CLASS_REGISTRY
+
+
+class SubdailyPModel:
+    r"""Fit a P Model incorporating fast and slow photosynthetic responses.
+
+    The :class:`~pyrealm.pmodel.pmodel.PModel` implementation of the P Model assumes
+    that plants instantaneously adopt optimal behaviour, which is reasonable where the
+    data represents average conditions over longer timescales and the plants can be
+    assumed to have acclimated to optimal behaviour. Over shorter timescales, this
+    assumption is unwarranted and photosynthetic slow responses need to be included.
+    This class implements the weighted-average approach of {cite:t}`mengoli:2022a`, but
+    is extended to include the slow response of :math:`\xi` in addition to
+    :math:`V_{cmax25}` and :math:`J_{max25}`.
+
+    The first dimension of the data arrays use to create the
+    :class:`~pyrealm.pmodel.pmodel_environment.PModelEnvironment` instance must
+    represent
+    the time
+    axis of the observations. The actual datetimes of those observations must then be
+    used to initialiase a :class:`~pyrealm.pmodel.fast_slow_scaler.FastSlowScaler`
+    instance, and one of the ``set_`` methods of that class must be used to define an
+    acclimation window.
+
+    The workflow of the model is then:
+
+    * The daily acclimation window set in the
+      :class:`~pyrealm.pmodel.fast_slow_scaler.FastSlowScaler` instance is used to
+      calculate daily average values for forcing variables from within the acclimation
+      window.
+    * A standard P Model is then run on those daily forcing values to generate predicted
+      states for photosynthetic parameters that give rise to optimal productivity.
+    * The :meth:`~pyrealm.pmodel.subdaily.memory_effect` function is then used to
```

Comment:
Maybe we can also mention the memory effect equation here for convenience of reading.

Reply:
Done.